File: TiketRegular.java
public class TiketRegular extends Tiket {
    // Properti tambahan khusus tiket Regular
    private String tipeAudio;
    private String lokasiBaris;

    // Constructor
    public TiketRegular(int idTiket, String namaFilm, String jadwalTayang, int jumlahKursi, int hargaDasarTiket, String tipeAudio, String lokasiBaris) {
        // 'super' digunakan untuk memanggil constructor dari kelas induk (Tiket)
        super(idTiket, namaFilm, jadwalTayang, jumlahKursi, hargaDasarTiket);
        this.tipeAudio = tipeAudio;
        this.lokasiBaris = lokasiBaris;
    }

    // Wajib meng-override (menimpa) metode abstrak dari kelas induk
    @Override
    public double hitungTotalHarga() {
        // Untuk sementara kita kembalikan harga dasarnya dulu
        return hargaDasarTiket; 
    }

    @Override
    public void tampilkanInfoFasilitas() {
        System.out.println("--- Fasilitas Studio Regular ---");
        System.out.println("Tipe Audio   : " + tipeAudio);
        System.out.println("Lokasi Baris : " + lokasiBaris);
    }
}

File: TiketRegular.php
<?php
require_once 'Tiket.php';

class TiketRegular extends Tiket {
    private $tipe_audio;
    private $lokasi_baris;

    public function __construct($id_tiket, $nama_film, $jadwal_tayang, $jumlah_kursi, $harga_dasar_tiket, $tipe_audio, $lokasi_baris) {
        // Memanggil constructor induk
        parent::__construct($id_tiket, $nama_film, $jadwal_tayang, $jumlah_kursi, $harga_dasar_tiket);
        $this->tipe_audio = $tipe_audio;
        $this->lokasi_baris = $lokasi_baris;
    }

    // Overriding Tahap 5
    public function hitungTotalHarga() {
        return $this->jumlah_kursi * $this->harga_dasar_tiket;
    }

    public function tampilkanInfoFasilitas() {
        return "Audio: {$this->tipe_audio} | Baris: {$this->lokasi_baris}";
    }
}
?>
